### PR TITLE
Refactor(eos_designs): Minor code adjustments caught by latest pylint

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Minor code adjustments caught by latest pylint

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/ip_addressing/utils.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ def _get_p2p_ipv4_pool_and_offset(self: "AvdIpAddressing", uplink_switch_index:
                 "Unable to assign IPs for uplinks. Either 'uplink_ipv4_pool' on this switch or 'downlink_pools' on all the uplink switches"
             )
 
-        if downlink_pool is None:
+        if uplink_pool is not None:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/ip_addressing/utils.py:178]: hmmm it was downlink_pool

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/ip_addressing/utils.py:178]: Check the logic. I an also doing _not_

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/ip_addressing/utils.py:178]: basically the function has two modes. Either downlink or uplink. Never both or none (checked and raised above).

**Review by gmuloc — APPROVED** [review]
LGTM

**Review by Shivani-gslab — APPROVED** [review]
LGTM

**Pull request merged**